godtools/App/Share/Data/StringSearcher/StringSearcher.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  StringSearcher.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 11/7/23.
+//  Copyright © 2023 Cru. All rights reserved.
+//
+
+import Foundation
+
+class StringSearcher {
```

Comment:
I initially thought I'd do dependency inversion here, but because it wasn't a repository it didn't fit the pattern we had talked about with dependency inversion.  So my followup question would be-- should we be using dependency inversion for _any_ type of dependencies? (any type of dependencies within use cases?)

Reply:
That's a good question for this instance here, originally I was thinking we would use di for any UseCase dependency, but since StringSearcher contains some business logic we need, now I'm wondering if we should just instantiate it directly in the UseCase itself.  It seems like we would need this logic included for behavior testing.

Reply:
It'd be good to write some behavior tests for the app languages list searching and tool category and language filtering to see if it's best to instantiate it directly in the UseCase, but that can be done in a separate ticket.

Reply:
With dependency inversion I really think it can be used anywhere.  I grabbed a quick definition of it here (https://www.geeksforgeeks.org/dependecy-inversion-principle-solid/)

```Any higher classes should always depend upon the abstraction of the class rather than the detail```

It fits really nicely into our behavior tests because we can easily mock the data since we only care about testing business rules and logic and aren't concerned about the data itself.

I think there are negatives as well to adding too much DI.  With more interfaces you add more abstraction which can make large projects harder to figure out as you sift through code trying to find the concrete implementation.

Reply:
Yeah, I do worry that having so many files makes the project difficult to navigate.  Made a ticket for writing those behavior tests:  https://jira.cru.org/browse/GT-2208

Reply:
Thanks @rachaelblue for creating that ticket.

I was hesitant about adding the domain layer because of the added files when using di.  However, I do like the behavior tests and my hope is they will be really helpful in creating a source of truth as to how the app should behave and function.

With di the primary focus will be in the UseCase layer.